sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -80,6 +80,20 @@ export class Sender {
     return sender.send(message);
   }
 
+  /**
+   * Ensures that the Service Bus connection and sender link is open.
+   *
+   * This method will be called on-demand by the Sender. Only call this method if you need
+   * to force the opening of the connection and link.
```

Comment:
This needs some work... I'd like us to convey that this is optional, not everyone should feel the need to use it. And a guidance on when one should consider calling it.

And also update our send code snippet in our readme and/or samples with something like the below:

> Add `await sender.open()` before your first send operation if you want to front load the work of setting up the AMQP link to the service. Doing this will save your first send operation from spending the time to do the same

Reply:
Agree with above, but it's `AMQP connection and link` I think?

Reply:
well, not always.

If you have used the client for any of the non send operations, then the AMQP connection is already up, all we need to do is set up the link

Reply:
In the interest of not going too much details, I would suggest to sticking to "link" only.

I would suggest the below for the docs on this public method

> Opens the AMQP link to Azure Service Bus from the sender. It is not necessary to call this method in order
to use the sender. It is recommended to call this before your first send() or sendBatch() calls if you want to front 
load the work of setting up the AMQP link to the service. Doing this will save your first send operation from 
spending the time to do the same.

Reply:
Updated, adding in some formatting.

I did remove the `Doing this will save...` part only because it seemed redundant with the prior sentences.

Reply:
(unresolving - didn't get to the sample yet)

Reply:
@ramya-rao-a  - pre-critique (I don't want to push again and have my tests stop!)

(this goes into the ### Send messages snippet):

// Optionally, you can call `sender.open()` if you want to front load the work of setting
// up the AMQP link to the service. If not called, the `sender` will call `open()` on your behalf
// on the first send operation.
//
// await sender.open()

Reply:
"Optionally, you can call `sender.open()`" -> "Optionally, you can await on `sender.open()` "to double down that this is an async operation or "Optionally, you can call `await sender.open()`"

"on the first send operation." -> "in the first send operation." ?